packages/dev/serializers/src/USD/USDSerializer.ts

```diff
@@ -0,0 +1,710 @@
+import { VertexBuffer } from "core/Buffers/buffer";
+import { Camera } from "core/Cameras/camera";
+import { PBRMaterial } from "core/Materials/PBR/pbrMaterial";
+import { Texture } from "core/Materials/Textures/texture";
+import { DynamicTexture } from "core/Materials/Textures/dynamicTexture";
+import { type Color3, type Matrix } from "core/Maths/math";
+import { Vector2 } from "core/Maths/math";
+import { type AbstractMesh } from "core/Meshes/abstractMesh";
+import { type TransformNode } from "core/Meshes/transformNode";
+import { type Scene } from "core/scene";
+import { strToU8, zipSync } from "fflate";

```

Comment:
Not sure actually, how would you do your idea?

Reply:
creating an async init function that loads these dependencies async. So the dev needs to actively "prepare" the library, or a similar architecture we have with other dependencies (like a physics engine) - inject the module during construction (expecting the dev to do the work, avoiding adding forced dependencies to babylon)

Reply:
So what should I do wit this then?  Leave it in the package or...?

Reply:
Avoid the dependency in package.json, and create a static init function that takes the (initiated) dependency. You could also add it to the constructor, if it is expected that the user will construct this class directly.

Reply:
Of course, you can use the types! and then the dependency will not be included in the build but typescript will support it.

Reply:
I actually don't know how to do that.

Reply:
if you leave the dependency (even as a dev-dependency) and use `import type`  instead of `import` , you will be able to define the object you expect from the developer to inject.

Something like this (very pseudo-style):

```
import type { strToU8, zipSync } from "fflate";

......

function Init(optionalfflate: { strToU8, zipSync }) {
// init the serializer
}
```

Let me talk to them team about it and see if we can simply keep the dependency instead of making this workaround. I'm keeping this open for now.

Reply:
ok! we had a long discussion. In regards to this - let's continue with the injection method instead of importing the dependency. The serializers should have that as an optional dependency and it should be documented that these two functions must be injected in order to create a serializer. As part of the parameters that are passed to the serialize function you should also provide an object with these 2 functions.

You can see the way the Havok plugin is doing it (whereas, in this case, it is a part of the constructor). As long as these functions are available when they are needed and provided by the developer, I am fine with the dependency.

Reply:
I honestly have no clue to do what you are asking for. :(

Reply:
you can check how havok was integrated as an external dependency, how cannon, ammo or recast are integrated.

The idea is to inject the functions and not include them as dependency.